hls/decode.go

```diff
@@ -11,27 +11,35 @@ import (
 //Parse reads a Master Playlist file and converts it to a MasterPlaylist object
 func (p *MasterPlaylist) Parse(reader io.Reader) error {
 	buf := manifest.NewBufWrapper()
-	buf.ReadFrom(reader)
-	if buf.Err != nil {
-		return buf.Err
+
+	// Populate buffer into memory, we could change this to a tokenizer with a line-by-line scanner?
+	if _, err := buf.ReadFrom(reader); err != nil {
+		return err
 	}
+
+	// Parsed Data
+	var renditions []*Rendition
+	variant := &Variant{}
+
+	// Parsing temporary state variables
 	var eof bool
+	// Was #EXTINF the last thing we checked? If so, we are looking for a URI line next
 	var streamInfLastTag bool
+	// Raw line
 	var line string
-	var renditions []*Rendition
 	key := &Key{}
-	variant := &Variant{}
 	r := &Rendition{}
 
-	//until EOF, read every line and decode into an object
+	// Runs until io.EOF, reads line-by-line from buffer and decode into an object
 	for !eof {
 		line = buf.ReadString('\n')
+		if buf.Err == io.EOF {
+			buf.Err = nil
```

Comment:
A change so that we don't return io.EOF on the Parse methods. Since a user should expect that the playlist should eventually end, I've cleared that so it isn't considered an error.